pkg/kotsutil/kots.go

```diff
@@ -564,6 +570,380 @@ func LoadKotsKindsFromPath(fromDir string) (*KotsKinds, error) {
 	return &kotsKinds, nil
 }
 
+// KotsKindsBundle is used for gathering rendered KOTS Kinds manifests for writing.
+type KotsKindsBundle struct {
+	KotsApplication *KotsKindsFile
+	Application     *KotsKindsFile
+	HelmCharts      []*KotsKindsFile
+
+	Collector     *KotsKindsFile
+	Preflight     *KotsKindsFile
+	Analyzer      *KotsKindsFile
+	SupportBundle *KotsKindsFile
+	Redactor      *KotsKindsFile
+	HostPreflight *KotsKindsFile
+
+	Config       *KotsKindsFile
+	ConfigValues *KotsKindsFile
+
+	Installation *KotsKindsFile
+	License      *KotsKindsFile
+
+	Identity       *KotsKindsFile
+	IdentityConfig *KotsKindsFile
+
+	Backup    *KotsKindsFile
+	Installer *KotsKindsFile
+
+	LintConfig *KotsKindsFile
+}
+
+func (b *KotsKindsBundle) Files() []*KotsKindsFile {
+	files := []*KotsKindsFile{}
+
+	if b.KotsApplication != nil {
+		files = append(files, b.KotsApplication)
+	}
+
+	if b.Application != nil {
+		files = append(files, b.Application)
+	}
+
+	for _, chart := range b.HelmCharts {
+		files = append(files, chart)
+	}
+
+	if b.Collector != nil {
+		files = append(files, b.Collector)
+	}
+
+	if b.Preflight != nil {
+		files = append(files, b.Preflight)
+	}
+
+	if b.Analyzer != nil {
+		files = append(files, b.Analyzer)
+	}
+
+	if b.SupportBundle != nil {
+		files = append(files, b.SupportBundle)
+	}
+
+	if b.SupportBundle != nil {
+		files = append(files, b.SupportBundle)
+	}
+
+	if b.Redactor != nil {
+		files = append(files, b.Redactor)
+	}
+
+	if b.HostPreflight != nil {
+		files = append(files, b.HostPreflight)
+	}
+
+	if b.Config != nil {
+		files = append(files, b.Config)
+	}
+
+	if b.ConfigValues != nil {
+		files = append(files, b.ConfigValues)
+	}
+
+	if b.Installation != nil {
+		files = append(files, b.Installation)
+	}
+
+	if b.License != nil {
+		files = append(files, b.License)
+	}
+
+	if b.Identity != nil {
+		files = append(files, b.Identity)
+	}
+
+	if b.IdentityConfig != nil {
+		files = append(files, b.IdentityConfig)
+	}
+
+	if b.Backup != nil {
+		files = append(files, b.Backup)
+	}
+
+	if b.Installer != nil {
+		files = append(files, b.Installer)
+	}
+
+	if b.LintConfig != nil {
+		files = append(files, b.LintConfig)
+	}
+
+	return files
+}
+
+func (b *KotsKindsBundle) KotsKinds() (*KotsKinds, error) {
```

Comment:
this can be replaced with a modified version of the `LoadKotsKindsFromPath` function

Reply:
👍 I broke that functionality out into a KotsKind method